src/libsyntax/parse/mod.rs

```diff
@@ -86,9 +90,13 @@ pub struct Directory {
 
 #[derive(Copy, Clone)]
 pub enum DirectoryOwnership {
-    Owned,
+    Owned {
+        // None if `mod.rs`, `Some(path)` if we're in
+        // `foo.rs` and there's a corresponding `foo/...`
```

Comment:
This comment makes it sound like we check for the existence of such a directory, but we don't right? Maybe "`Some("foo")` if we're in `foo.rs`?